docs/specs/clients/notify/authentication.md

```diff
@@ -175,6 +177,136 @@ Paginated list of notifications with the most recently sent first. Unread notifi
 | Tag     | 4015         |
 | Topic   | notify topic |
 
+## `wc_notifyNotificationChanged`
+
+Emitted by the Notify Server when a notification changed state. For example if its read status changed.
+
+### Request
+
+- act - `notify_notification_changed`
+- iss - did:key of dapp authentication key
+- aud - did:key of client identity key
+- nfn - array of [Notify Notification](./data-structures.md#notify-notification)
+
+```typescript
+{
+  auth: string,
+}
+```
+
+| IRN     |              |
+| ------- | ------------ |
+| TTL     | 300          |
+| Tag     | 4018         |
+| Topic   | notify topic |
+
+### Response
+
+- act - `notify_notification_changed_response`
+- iss - did:key of client identity key
+- ksu - key server for identity key verification
+- aud - did:key of dapp authentication key
+
+```typescript
+{
+  auth: string,
+}
+```
+
+| IRN     |              |
+| ------- | ------------ |
+| TTL     | 300          |
+| Tag     | 4019         |
+| Topic   | notify topic |
+
+## `wc_notifyReadNotification`
+
+Marks a notification as read.
+
+### Request
+
+- act - `notify_read_notification`
+- iss - did:key of client identity key
+- ksu - key server for identity key verification
+- aud - did:key of dapp authentication key
+- app - did:web of app domain that this request is associated with 
+  - Example: `did:web:app.example.com`
+- ids - array of notification IDs to mark as read, max 1000 items
```

Comment:
Add flag to mark all messages as read

Reply:
Decided not to do this

Reply:
I think we should actually do a flag to mark all as read. Doing this all client-side depends on pagination as you can only `markAllReturnedNotificationsAsRead` and so for long notification lists the user would never mark them all as read.

Reply:
Added in [57dfec4](https://github.com/WalletConnect/walletconnect-specs/pull/191/commits/57dfec439bc52a1e23a14939a3270337c663334c)